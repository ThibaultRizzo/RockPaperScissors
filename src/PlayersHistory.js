import React, { Component } from 'react';

class PlayersHistory extends Component {
  render() {
    return (
      <div cid="PlayersHistory">
        <h1>Players History</h1>;
      </div>
    );
  }
}

export default PlayersHistory;
