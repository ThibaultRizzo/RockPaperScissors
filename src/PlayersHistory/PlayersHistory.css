#PlayersHistory {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: stretch;
}

.game-row {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-around;
  align-items: center;
}

.win-rate {
  align-self: center;
  margin-top: 20px;
  font-weight: bold;
}
