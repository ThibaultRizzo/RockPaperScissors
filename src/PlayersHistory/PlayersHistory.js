import React, { Component } from 'react';
import './PlayersHistory.css';

class PlayersHistory extends Component {
  render() {
    let games;
    games = this.props.gameHistory.map((game, i) => {
      let details = game[game.length - 1];
      return (
        <div className="game-row" key={i}>
          <div>{i + 1}</div>
          <div>{details.result}</div>
          <div>{details.scoreP1 + ' - ' + details.scoreP2}</div>
        </div>
      );
    });
    return (
      <div id="PlayersHistory">
        <h1>Players History</h1>
        {games}
      </div>
    );
  }
}

export default PlayersHistory;
