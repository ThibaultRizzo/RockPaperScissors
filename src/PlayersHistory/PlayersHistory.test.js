import React from 'react';
import ReactDOM from 'react-dom';
import PlayersHistory from './PlayersHistory';
import renderer from 'react-test-renderer';

const games = [
  [
    { moveP1: 'SCISSORS', moveP2: 'ROCK', result: 'LOSE' },
    { moveP1: 'SCISSORS', moveP2: 'ROCK', result: 'LOSE' },
    { moveP1: 'SCISSORS', moveP2: 'SCISSORS', result: 'DRAW' },
    {gameType
      :
      "AI vs AI", result: 'LOSE', scoreP1: 0, scoreP2: 2 }
  ],
  [
    { moveP1: 'PAPER', moveP2: 'ROCK', result: 'WIN' },
    { moveP1: 'SCISSORS', moveP2: 'ROCK', result: 'LOSE' },
    { moveP1: 'SCISSORS', moveP2: 'ROCK', result: 'LOSE' },
    { moveP1: 'PAPER', moveP2: 'PAPER', result: 'DRAW' },
    { moveP1: 'ROCK', moveP2: 'SCISSORS', result: 'WIN' },
    { gameType
      :
      "Player vs AI",result: 'DRAW', scoreP1: 2, scoreP2: 2 }
  ],
  [
    { moveP1: 'ROCK', moveP2: 'SCISSORS', result: 'WIN' },
    { moveP1: 'PAPER', moveP2: 'SCISSORS', result: 'LOSE' },
    { moveP1: 'PAPER', moveP2: 'SCISSORS', result: 'LOSE' },
    { moveP1: 'PAPER', moveP2: 'PAPER', result: 'DRAW' },
    { gameType
      :
      "Player vs AI",result: 'LOSE', scoreP1: 1, scoreP2: 2 }
  ]
];

it('renders correctly', () => {
  const rendered = renderer.create(<PlayersHistory gameHistory={games} />);
  expect(rendered.toJSON()).toMatchSnapshot();
});
