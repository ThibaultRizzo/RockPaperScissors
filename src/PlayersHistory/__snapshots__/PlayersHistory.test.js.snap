// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="PlayersHistory"
>
  <h1>
    Players History
  </h1>
  <div
    className="game-row lose"
  >
    <div>
      1
    </div>
    <div>
      LOSE
    </div>
    <div>
      0 - 2
    </div>
  </div>
  <div
    className="game-row draw"
  >
    <div>
      2
    </div>
    <div>
      DRAW
    </div>
    <div>
      2 - 2
    </div>
  </div>
  <div
    className="game-row lose"
  >
    <div>
      3
    </div>
    <div>
      LOSE
    </div>
    <div>
      1 - 2
    </div>
  </div>
  <div
    className="win-rate"
  >
    Win Rate: 
    0%
  </div>
</div>
`;
